spring-cloud-huawei-servicecomb-discovery/src/main/java/com/huaweicloud/servicecomb/discovery/client/ServiceCombClient.java

```diff
@@ -411,12 +411,13 @@ public HeardBeatStatus heartbeat(HeartbeatRequest heartbeatRequest) throws Servi
       if (response.getStatusCode() == HttpStatus.SC_OK) {
         LOGGER.debug("heartbeat success.");
         return HeardBeatStatus.SUCCESS;
-      } else {
+      } else if (response.getStatusCode() == HttpStatus.SC_BAD_REQUEST) {
         LOGGER.error(
             "heartbeat to service center failed. status:" + response.getStatusCode() + "; message:"
                 + response.getStatusMessage() + "; content:" + response.getContent());
         return HeardBeatStatus.FAILED;
       }
+      throw new IOException();
```

Comment:
This exception is caught in the method, throw RemoteOperationException is more direct.

Reply:
The purpose of throwing an exception here is to trigger toggle() to back off.